src/Code/201908_draft/Client/Alfred.js

```diff
@@ -36,11 +35,10 @@ const WORKFLOW_PAUSE = "pause";
 const WORKFLOW_RUN = "run";
 
 const Alfred = class Alfred {
-  constructor(appId, apiToken, appEnv) {
-    /* This was moved in a singleton module because whatever client is used to
-     * init the credentials, they need to be shared between all code paths
-     * clients */
-    init(appId, apiToken, appEnv);
+  constructor(credentials) {
+    this.appId = credentials.appId;
+    this.apiToken = credentials.apiToken;
+    this.appEnv = credentials.appEnv;
   }
```

Comment:
If I'm not wrong, a copy of the value are done here. So if - for any reason - the user reuses the client and change its parameters (eg. to send an event to a different app in a sequential way) - it won't correctly be taken into account here. For this reason, It seems natural to me to use the Client object, not its internal values

Reply:
Ok, changes done.